### DISCUSSION TITLE
How to limit/downscale a video resolution output in iw3?

### DISCUSSION THREAD

**Discussion by MaryAtlas** [nagadomi/nunif#227] [Q&A] [answered]
Hello.
I am using iw3 to convert a 2D 4K resolution video to VR. I ended up with a 12k resolution video. I don't have a device capable of 12k playback.
Is there a way or setting in iw3 to give me a 8k resolution output? I know I could downscale the video first or after converting, but I don't want to re-encode the video twice, losing in data quality further. I prefer for iw3 to do everything together, and re-encode them only once.

**Comment by nagadomi** [comment]
When `StereoFormat=Half SBS` is specified, the output video will have the same resolution as the input video.
However, the horizontal resolution is reduced to 1/2.
Also, Half SBS support in the video player or the ability to manually change the aspect ratio is required.

Another method is to pre-downscale the input video using `-vf(src)` filter.
For example, setting `scale=-1:1080` in `-vf(src)` field will downscale the input video to `height=1080px` before processing.

Also, if the reason for not being able to playback is bitrate, you can reduce the filesize by specifying `VideoCodec=libx265` and Preset=`medium`. https://github.com/nagadomi/nunif/issues/203#issuecomment-2293803303

**Reply by MaryAtlas** [reply]
Pardon me I forgot to state the Stereo format I use. I use VR90, the rest did not do 3D well with my video player on SBS180 orientation(Video became stretched the closer it is to the border, fishbowl view, etc).
Thank you, I will try the -vf(src) filter. Also, thank you for the tip on bitrate. My VR device is Pico4. Its specs says it is limited to 8k, but I wonder now if it could play higher resolution if the bitrate is lowered.

**Reply by nagadomi** [reply]
Using VR90 for normal video viewing is not recommended.
I don't know about pico4, but most video players should support multiple 3D format (but automatic format detection may not work).
It is better to check the display mode setting of the video player to see if it really does not support SBS (Flat 3D).

**Reply by MaryAtlas** [reply]
I tried scale=-1:1080 and it worked. I have a question, if I set it to a strange value, say scale=-1:2700, will the video keep its original proportion?

**Reply by nagadomi** [reply, marked as answer]
`scale` filter is `scale=width:height` format, and when `-1` is specified , it is automatically set to keep the original aspect ratio.
However, if the size of width or height is not multiple of 2, h264 encoding error will occur.
Edit: When `-2` is specified, the (width) value is automatically adjusted to a multiple of 2 while maintaining the aspect ratio.

some examples can be found on the following pages
https://trac.ffmpeg.org/wiki/Scaling

**Reply by MaryAtlas** [reply]
I got what I wanted for the output using scale filter. Thank you very much.

**Comment by diverswan9** [comment]
I was recently trying this using a 4k source and did the -vf(src) field trick. I just put in  `scale=1920:1080`  (same effect as `scale=-1:1080`)
The nice thing about that is that it processes it as 1080 and not 2160 so it doesn't compromise speed.

**Reply by MaryAtlas** [reply]
Thanks for the tip. I will try it now and see what I get.

**Reply by diverswan9** [reply]
@MaryAtlas  - you need the 'source' to be 2D-180 for the VR90 to display properly in VR as 3D-180. If your source is just a regular 2D video(or image) it won't work to view as a 3D-180 VR video but just a regular 3D-SBS video. Does that make sense?
2D 180 images are usually taken with fish-eye lens cameras (they have a wide field of view)

The 2D 180 photo looks like this...
http://www.paulbourke.net/dome/cameras/fish4.png

**Reply by MaryAtlas** [reply]
I get what you mean. You mean to become VR, you need that fish-eye similar effect there, correct?
But the thing is, didn't you notice the VR90 option did that a little? The output of VR90 has that slight curve that curves the video, and it really works and I think it is the best you will get when you only have a regular 2D to work with.

**Reply by nagadomi** [reply]
@diverswan9 
VR90 option converts regular 2D to VR180 SBS (but it only provides 90 degrees of frontal view).
It looks almost same as Full SBS, but the screen position is not adjustable and the video resolution is about 3 times larger (most areas are only black borders).
